jax/numpy/lax_numpy.py

```diff
@@ -545,6 +545,28 @@ def exp2(x):
   return lax.exp(lax.mul(lax.log(_constant_like(x, 2)), x))
 
 
+@_wraps(onp.signbit)
+def signbit(x):
+  x, = _promote_shapes("signbits", x)
+
+  if issubdtype(_dtype(x), onp.integer):
+    x = lax.convert_element_type(x, onp.float64)
+
+  info = finfo(_dtype(x))
+
+  if info.bits == 16:
+    int_type = onp.int16
```

Comment:
One more slight nit: it's probably better to use unsigned integer types here. At least in C, the behavior of right-shifts of signed types is implementation-defined. (Not sure about NumPy, but I doubt it is different.) For unsigned types right shifts are always well-defined.

Reply:
Not sure why, but If I cast them to unsigned int, it'll throw the following errors
```sh
shift_right_arithmetic does not accept dtype float64 at position 0. Accepted dtypes at position 0 are subtypes of int64.
```

Reply:
That seems odd, and note the error: it looks like you must have been passing a `float64` as argument, not a `uint64`.

That said, I don't think it's worth worrying about.

Reply:
@hawkinsp Thanks for the feedback! I find that only the error only occurs in the following case.

```python
import jax.numpy as np
import numpy as onp
from jax import lax

x = onp.array([1.0, 2.0, 3.0], onp.float64)
y = lax.bitcast_convert_type(x, onp.uint64)
y >> 1
```

If I do the same thing on `float32`/`uint32`, it won't fail.